crossbeam-skiplist/src/base.rs

```diff
@@ -227,6 +227,24 @@ impl<K, V> Node<K, V> {
         }
     }
 
+    /// Decrements the reference count of a node, pinning the thread and destoying the node
+    /// if the count become zero.
+    #[inline]
+    unsafe fn decremnt_with_pin<F>(&self, pin: F)
+    where
+        F: FnOnce() -> Guard,
+    {
+        if self
+            .refs_and_height
+            .fetch_sub(1 << HEIGHT_BITS, Ordering::Release)
+            >> HEIGHT_BITS
+            == 1
+        {
+            fence(Ordering::Acquire);
+            pin().defer_unchecked(move || Self::finalize(self));
```

Comment:
Function `release()` has this `check_guard` call that will make sure the passed guard is the right one (belongs to the same GC instance the parent skip list uses). We need to do the same check here:

```rust
let guard = &pin();
parent.check_guard(guard);
defer_unchecked(move || Self::finalize(self));
```

Note that we'll also need to pass a reference to the parent skip list as an argument.

Reply:
Thanks! Added a call to `check_guard` in https://github.com/crossbeam-rs/crossbeam/pull/337/commits/46c8b530b528eef27f86ded3c153e2ed3650e159.